ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -1218,7 +1221,7 @@ boolean getIndexEntriesSince(StoreKey key, FindEntriesCondition findEntriesCondi
         tempMap = index.tailMap(key, true);
       }
       for (Map.Entry<StoreKey, ConcurrentSkipListSet<IndexValue>> entry : tempMap.entrySet()) {
-        if (key == null || entry.getKey().compareTo(key) != 0) {
+        if (key == null || entry.getKey().compareTo(key) != 0 || inclusive) {
```

Comment:
yeah, this is really good suggestion!

Reply:
I just wonder why previous code was written in that way. Looks like it could change `tempMap = index.tailMap(key, false);` to remove "if" condition in the for loop.